addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -552,15 +552,17 @@ def get_allocation_data(self, employees, target_date=None):
                                                                         )
                 if closest_expiration_date:
                     closest_allocation_expire = format_date(self.env, closest_expiration_date)
-                    calendar = employee.resource_calendar_id\
-                                or self.env.company.resource_calendar_id
-                    # closest_allocation_duration corresponds to the time remaining before the allocation expires
-                    calendar_attendance = calendar._work_intervals_batch(
-                        datetime.combine(target_date, time.min).replace(tzinfo=pytz.UTC),
-                        datetime.combine(closest_expiration_date, time.max).replace(tzinfo=pytz.UTC),
-                        resources=employee.resource_id
-                    )
-                    closest_allocation_dict = calendar._get_attendance_intervals_days_data(calendar_attendance[employee.resource_id.id])
+                    calendar = employee.resource_calendar_id
+                    start_date = datetime.combine(target_date, time.min).replace(tzinfo=pytz.UTC)
+                    end_date = datetime.combine(closest_expiration_date, time.max).replace(tzinfo=pytz.UTC)
```

Comment:
Nitpicking for clarity :innocent: 
```suggestion
                    start_datetime = datetime.combine(target_date, time.min).replace(tzinfo=pytz.UTC)
                    end_datetime = datetime.combine(closest_expiration_date, time.max).replace(tzinfo=pytz.UTC)
```